### DISCUSSION TITLE
open file in gui and change file in gui via xplr

### DISCUSSION THREAD

**Discussion by idrisr** [sayanarijit/xplr#470] [Q&A]
I have a directory full of pdfs, and I want to open each pdf, one at a time, in an external program like okular.  I am able to open the pdf with `gx`, but then to view the next pdf, I have to to quit the external program and then open it again for the next pdf.

Also xplr does not respond to any keystrokes when okular is open with the pdf.  Is there a way to leave the external program open, and change which pdf it is showing through xplr?

**Comment by sayanarijit** [comment]
Please provide more info. Like what system are you using? What's your desktop environment? Terminal?
It works fine on my Manjaro KDE+KWin. FYI, You can always edit the key bindings to use your choice of program to open files.
Docs: See https://xplr.dev/en/configure-key-bindings and [init.lua](https://github.com/sayanarijit/xplr/blob/main/src/init.lua).

**Reply by idrisr** [reply]
Arch Linux x86_64, zsh 5.8.1 and tmux 3.2a, and the terminal is Alacritty. I'm using the Awesome window manager (v3.4) with no desktop environment.  Thanks!

**Reply by sayanarijit** [reply]
Hmm... It will take some time for me to reproduce it in AwesomeWM. I'd recommend you to post it in the matrix or discord chat room (both are bridged). There are users using AwesomeWM who could probably help faster.

**Reply by sayanarijit** [reply]
https://xplr.dev/en/community

**Reply by sayanarijit** [reply]
Another option is to use the [preview-tabbed.xplr](https://github.com/sayanarijit/preview-tabbed.xplr) plugin, which also displays pdf.

**Reply by erikLundstedt** [reply]
Hi
I'm using a very similar environment (arch+zsh+awesome)
But I sadly don't have any idea at the moment on how to do what you want


Could you provide your xplr lua(or fennel) configuration code?


What happens if you try to do the same thing using something like fzf using the preview feature?
(If you try this, please provide the code for that as well)

**Reply by sayanarijit** [reply]
@idrisr

**Reply by idrisr** [reply]
The `init.lua` I'm using has only the minimum 
```
version="0.17.4"
```

Is this the fzf preview feature you're referring to? https://github.com/sayanarijit/fzf.xplr

**Reply by idrisr** [reply]
I installed the `fzf.xplr` plugin, and that works fine for text files, after I figured out I needed to replace the `pistol` file viewer with `bat`.  It doesn't do anything useful for pdfs, as a pdf needs a non-terminal application like `okular` to launch.
